File: MovieCatalogueV2/app/src/main/java/com/google/fdp/moviecataloguev2/viewmodels/MovieViewModel.java
package com.google.fdp.moviecataloguev2.viewmodels;

import android.app.Application;

import androidx.annotation.NonNull;
import androidx.lifecycle.AndroidViewModel;
import androidx.lifecycle.LiveData;

import com.google.fdp.moviecataloguev2.models.Movie;
import com.google.fdp.moviecataloguev2.repositories.MovieRepository;

import java.util.List;

/**
 * Created by gama on 2019-07-27.
 * Addin Gama Bertaqwa
 * addingama@gmail.com
 */
public class MovieViewModel extends AndroidViewModel {

    private MovieRepository movieRepository;

    public MovieViewModel(@NonNull Application application) {
        super(application);
        movieRepository = MovieRepository.getInstance(application);
    }


    public LiveData<List<Movie>> getLiveData() {
        return movieRepository.getAllMovies();
    }

    public void updateItem(String type) {
        movieRepository.updateData(type);
    }



}

File: MovieCatalog/app/src/main/java/com/alfianh/moviecatalog/viewmodels/MovieViewModel.kt
package com.alfianh.moviecatalog.viewmodels

import android.app.Application
import androidx.lifecycle.AndroidViewModel
import androidx.lifecycle.LiveData
import androidx.lifecycle.viewModelScope
import com.alfianh.moviecatalog.models.Movie
import com.alfianh.moviecatalog.repository.MovieRepository
import kotlinx.coroutines.launch

class MovieViewModel constructor(application: Application) : AndroidViewModel(application) {

  private var movieRepository: MovieRepository = MovieRepository.getInstance(getApplication())

  fun getLiveData(): LiveData<List<Movie>> = movieRepository.getAllMovies()

  fun updateItem(type: String) = movieRepository.updateData(type)

  fun getAllFavorite(type: String) = movieRepository.getAllFavorite(type)

  fun insertFavorite(movie: Movie, type: String) = viewModelScope.launch {
    movieRepository.insertFavorite(movie, type)
  }

  fun deleteFavorite(movie: Movie, type: String) = viewModelScope.launch {
    movieRepository.deleteFavorite(movie, type)
  }

  fun isFavorite(id: Long, type: String) = movieRepository.isFavorite(id, type)

}